File: js/levelOrder.js
// INCOMPLETE, IMPLEMENTED INCORRECT TRAVERSAL
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if(root) {

        let a = levelOrder(root.left).concat([root.val]).concat(levelOrder(root.right))
        console.log(a)
        return a
    }
    console.log(root)

    return []
    
};

File: level_order.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        final_order = []
        curr_level_q = []
        todo_q = [root]
        level = 0
        first = True
        while len(todo_q) > 0:
            curr_level_q = todo_q
            level += 1
            todo_q = []
            while len(curr_level_q) > 0:
                node = curr_level_q.pop(0)
                if node:
                    if node.left:
                        todo_q.append(node.left)
                    if node.right:
                        todo_q.append(node.right)
                    if level > len(final_order):
                        final_order.append([node.val])
                    else:
                        final_order[level-1].append(node.val)
        return final_order

            